rascaline/tests/lode-madelung.rs

```diff
@@ -0,0 +1,103 @@
+fn get_zns() -> Vec<Box<dyn System>> {
+    let cell = Matrix3::new([[0.0, 1.0, 1.0], [1.0, 0.0, 1.0], [1.0, 1.0, 0.0]]);
+    let mut system = SimpleSystem::new(UnitCell::from(cell));
+    system.add_atom(16, Vector3D::new(0.0, 0.0, 0.0));
+    system.add_atom(30, Vector3D::new(0.5, 0.5, 0.5));
+    return vec![Box::new(system) as Box<dyn System>];
+}
+
+
+/// ZnS (O4) in wurtzite structure (triclinic cell)
+fn get_znso4() -> Vec<Box<dyn System>> {
+    let u = 3. / 8.;
+    let c = f64::sqrt(1. / u);
+    let cell = Matrix3::new([[0.5, -0.5 * f64::sqrt(3.0), 0.0], [0.5, 0.5 * f64::sqrt(3.0), 0.0], [0.0, 0.0, c]]);
+    let mut system = SimpleSystem::new(UnitCell::from(cell));
+    system.add_atom(16, Vector3D::new(0.5, 0.5 / f64::sqrt(3.0), 0.0));
+    system.add_atom(30, Vector3D::new(0.5, 0.5 / f64::sqrt(3.0), u * c));
+    system.add_atom(16, Vector3D::new(0.5, -0.5 / f64::sqrt(3.0), 0.5 * c));
+    system.add_atom(30, Vector3D::new(0.5, -0.5 / f64::sqrt(3.0), (0.5 + u) * c));
+    return vec![Box::new(system) as Box<dyn System>];
+}
+
+
+#[test]
+fn madelung() {
+    let mut crystals = [
+        CrystalParameters{systems: get_nacl(), charges: vec![1.0, -1.0], madelung: 1.7476},
+        CrystalParameters{systems: get_cscl(),  charges: vec![1.0, -1.0], madelung: 2.0 * 1.7626 / f64::sqrt(3.0)},
+        CrystalParameters{systems: get_zns(),  charges: vec![1.0, -1.0], madelung: 2.0 * 1.6381 / f64::sqrt(3.0)},
+        CrystalParameters{systems: get_znso4(),  charges: vec![1.0, -1.0, 1.0, -1.0], madelung: 1.6413 / f64::sqrt(3. / 8.)}
+    ];
+
+    for atomic_gaussian_width in [0.2, 0.1] {
+        for cutoff in [0.01, 0.027, 0.074, 0.2] {
+            for crystal in crystals.iter_mut() {
+
+                let lode_parameters = LodeSphericalExpansionParameters {
+                    cutoff: cutoff,
+                    k_cutoff: None,
+                    max_radial: 1,
+                    max_angular: 0,
+                    atomic_gaussian_width: atomic_gaussian_width,
+                    center_atom_weight: 0.0,
+                    potential_exponent: 1,
+                    radial_basis: LodeRadialBasis::splined_gto(1e-8),
+                };
+
+                let mut calculator = Calculator::from(Box::new(LodeSphericalExpansion::new(
+                    lode_parameters
+                ).unwrap()) as Box<dyn CalculatorBase>);
+
+                let options = CalculationOptions {..Default::default()};
+    
+                let descriptor = calculator.compute(&mut crystal.systems, options).unwrap();
+
+                let madelung = (
+                        crystal.charges[0] * descriptor.block_by_id(0).values().data.as_array()[[0, 0, 0]]
+                      + crystal.charges[1] * descriptor.block_by_id(1).values().data.as_array()[[0, 0, 0]])
+                    / -(4.0 * std::f64::consts::PI * cutoff.powf(2.0)).powf(0.75);
+
+                assert_relative_eq!(madelung, crystal.madelung, max_relative=8e-2);
```

Comment:
this is quite a high relative error. Can we make it go down by increasing the k_cutoff?

Reply:
The reason is that we probe a wide range of `cutoff`s and` atomic_gaussian_width`s. We could bring down the accuracy to `5e-4` if just restrict the test to the smallest `cutoff` and and `atomic_gaussian_width`...

Reply:
Could we have two tests/two assert_relative_eq? One which would probe with high accuracy, and the ones which are already here. This way we can still catch small errors if they start occurring later.

Reply:
Yes we can split.